app/src/main/java/org/fossasia/openevent/activities/TracksActivity.java

```diff
@@ -39,9 +46,11 @@ protected void onCreate(Bundle savedInstanceState) {
                 (CollapsingToolbarLayout) findViewById(R.id.collapsing_toolbar);
         collapsingToolbar.setTitle(track);
 
+
         RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
         try {
-            sessionsAdapter = new SessionsAdapter(dbSingleton.getSessionbyTracksname(track));
+            List<Session> sessionList = dbSingleton.getSessionbyTracksname(track);
```

Comment:
Doing this on the UI Thread may make the app slow, we can follow a pattern using observables RxJava or Loaders etc to make the transition as smooth as possible. 

This bit is advanced but its a useful thing to understand. http://blog.danlew.net/2015/06/22/loading-data-from-multiple-sources-with-rxjava/